src/scripts/main.js

```diff
@@ -1,3 +1,430 @@
+    pushNotification(
+      580,
+      10,
+      'Enter a correct name!',
+      'Name must contain at least 4 letters and not include numbers',
+      'error'
+    );
+
+    isValid = false;
+  } else if (data.get('age') < minAge
+  || data.get('age') > maxAge) {
+    pushNotification(
+      580,
+      10,
+      'Enter a correct age!',
+      'Your age must be from 18 to 90 years',
+      'error'
+    );
+
+    isValid = false;
+  } else if (data.get('position').trim().length < nameLength
+    || /\d/.test(data.get('position'))) {
+    pushNotification(
+      580,
+      10,
+      'Enter a position!',
+      'You should enter a correct position',
+      'error'
+    );
+
+    isValid = false;
+  } else if (data.get('salary') <= 0) {
+    pushNotification(
+      580,
+      10,
+      'Enter a correct salary!',
+      'Salary must be more than 0',
+      'error'
+    );
+
+    isValid = false;
+  } else {
+    isValid = true;
+  }
+};
+
+addEmployee.addEventListener('submit', submitEvent => {
+  const employeeInfo = new FormData(addEmployee);
+  const salaryToNormalize = employeeInfo.get('salary');
+  let normalizedSalary = salaryToNormalize;
+
+  if (salaryToNormalize.length > 3) {
+    normalizedSalary
+    = salaryToNormalize.slice(0, -3)
+    + ','
+    + salaryToNormalize.slice(-3, salaryToNormalize.length);
+  }
+
+  const salary = `$${normalizedSalary}`;
+
+  submitEvent.preventDefault();
+  validation(employeeInfo);
+
+  if (!isValid) {
+    return;
+  }
+
+  const newRow = tbody.insertRow();
+
+  newRow.insertCell(0).innerText = employeeInfo.get('name');
+  newRow.insertCell(1).innerText = employeeInfo.get('position');
+  newRow.insertCell(2).innerText = employeeInfo.get('office');
+  newRow.insertCell(3).innerText = employeeInfo.get('age');
+  newRow.insertCell(4).innerText = salary;
+
+  rows.push(newRow);
+
+  addEmployee.reset();
+
+  pushNotification(
+    460,
+    10,
+    'Validation success!',
+    'Congrats! New employee is successfully added into the table',
+    'success');
+});
+
+function pushNotification(posTop, posRight, title, description, type) {
+  const message = document.createElement('div');
+  const messageTitle = document.createElement('h2');
+  const messageDescription = document.createElement('p');
+
+  message.style.top = `${posTop}px`;
+  message.style.right = `${posRight}px`;
+
+  messageTitle.innerText = title;
+  messageDescription.innerText = description;
+
+  message.classList.add('notification', type);
+  message.setAttribute('data-qa', 'notification');
+  messageTitle.classList.add('title');
+
+  message.append(messageTitle, messageDescription);
+  document.body.append(message);
+
+  setTimeout(() => {
+    message.remove();
+  }, timeoutOfMessage);
+};
+
+// change, validate and save data in cell-input
+
+tbody.addEventListener('dblclick', doubleClick => {
+  const target = doubleClick.target.closest('td');
+  const input = document.querySelector('.cell-input');
```

Comment:
show the initial value in input:
![image](https://user-images.githubusercontent.com/86923089/219943129-5ef12e9e-e4cf-48fa-be4c-709f04cb2ceb.png)